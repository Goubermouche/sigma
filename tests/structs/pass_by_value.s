struct user {
	i32* k;
	i32 x;
};

void print(user usr) {
	printf("key: %d %d %d\n", usr.k[0], usr.k[1], usr.x);
}

i32 main() {
	user my_user;

	my_user.k = cast<i32*>(malloc(8));
	my_user.k[0] = 123;
	my_user.k[1] = 321;
	my_user.x = 111;
	
	printf("key: %d %d %d\n", my_user.k[0], my_user.k[1], my_user.x);
	print(my_user);

	ret 0;
}
