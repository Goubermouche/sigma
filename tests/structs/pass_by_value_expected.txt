key: 123 321 111
key: 123 321 222
key: 123 321 111
